sim/aviron/src/lib/mcu.zig

```diff
@@ -0,0 +1,200 @@
+const std = @import("std");
+const io_mod = @import("io.zig");
+const memory = @import("memory.zig");
+const Cpu = @import("Cpu.zig");
+
+/// MCU configuration defines the memory layout and special register addresses
+/// for a specific AVR microcontroller.
+pub const Config = struct {
+    /// Human-readable name of the MCU
+    name: []const u8,
+
+    /// Flash memory size in bytes (must be 2-aligned)
+    flash_size: usize,
+
+    /// SRAM size in bytes
+    sram_size: u16,
+
+    /// SRAM base address in data space
+    sram_base: u16,
+
+    /// EEPROM size in bytes
+    eeprom_size: usize,
+
+    /// Code model (determines PC width)
+    code_model: Cpu.CodeModel,
+
+    /// Instruction set variant
+    instruction_set: Cpu.InstructionSet,
+
+    /// Special I/O register addresses
+    special_io: SpecialIoConfig,
+
+    /// Start of IO window in data address space (inclusive).
+    io_window_base: u24,
+    /// End of IO window in data address space (inclusive).
+    io_window_end: u24,
+};
+
+/// Convenience container for constructed memory spaces.
+pub const Spaces = struct {
+    data: memory.MemorySpace,
+    io: memory.MemorySpace,
+    eeprom: memory.MemorySpace,
+
+    pub fn deinit(self: *const Spaces, alloc: std.mem.Allocator) void {
+        self.data.deinit(alloc);
+        self.io.deinit(alloc);
+        self.eeprom.deinit(alloc);
+    }
+};
+
+/// Build memory spaces (data, io, eeprom) for the given MCU configuration.
+pub fn build_spaces(
+    alloc: std.mem.Allocator,
+    cfg: Config,
+    sram_dev: io_mod.Device,
+    io_dev: io_mod.Device,
+    eeprom_dev: io_mod.Device,
+) !Spaces {
```

Comment:
There is no need to pass `sram_dev` and `eeprom_dev` here as they can be allocated from `cfg` directly with a flexibly sized memory.

Otherwise it's just more work for the programmer and a source of bugs when they don't match `cfg`.

Reply:
True, but this function lets the device to be fixed size or flexible, deferring to the caller to clean up the memory and avoiding runtime allocations when we can.

I propose I clean it up in the other PR, where I will add the dynamic support